tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsAmqpIT.java

```diff
@@ -119,37 +134,84 @@ public void testGetCredentialsNotExistingAuthId(final TestContext ctx) {
 
     /**
      * Verifies that the service returns credentials for a given type and authentication ID.
-     * 
+     *
      * @param ctx The vert.x test context.
      */
     @Test
     public void testGetCredentialsReturnsCredentialsTypeAndAuthId(final TestContext ctx) {
 
-        credentialsClient
-            .get(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, CREDENTIALS_AUTHID1)
-            .setHandler(ctx.asyncAssertSuccess(result -> {
-                ctx.assertEquals(CREDENTIALS_AUTHID1, result.getAuthId());
-                ctx.assertEquals(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, result.getType());
-            }));
+        // Prepare the credential to insert
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final JsonObject payload = new JsonObject()
+                .put(CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID, deviceId)
+                .put(CredentialsConstants.FIELD_TYPE, CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD)
+                .put(CredentialsConstants.FIELD_AUTH_ID, CREDENTIALS_AUTHID1)
+                .put(CredentialsConstants.FIELD_SECRETS, getAuthId1Secrets());
+
+        final Async done = ctx.async();
+        // Insert it into the device Registry
+        helper.registry.addCredentials(Constants.DEFAULT_TENANT, payload).setHandler(r -> {
+
+            // it's successfully inserted, run the test.
+            if (r.succeeded()) {
+                credentialsClient
+                        .get(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, CREDENTIALS_AUTHID1)
+                        .setHandler(ctx.asyncAssertSuccess(result -> {
+                            ctx.assertEquals(CREDENTIALS_AUTHID1, result.getAuthId());
+                            ctx.assertEquals(CredentialsConstants.SECRETS_TYPE_HASHED_PASSWORD, result.getType());
+                            done.complete();
+                        }));
+            }else {
+                ctx.fail(r.cause());
+            }
+        });
+        done.await();
     }
 
-    /**
-     * Verifies that the service returns credentials for a given type, authentication ID and matching client context.
-     *
-     * @param ctx The vert.x test context.
-     */
+/**
+ * Verifies that the service returns credentials for a given type, authentication ID and matching client context.
+ *
+ * @param ctx The vert.x test context.
+ */
     @Test
     public void testGetCredentialsExistingClientContext(final TestContext ctx) {
 
+        // Prepare the credential to insert
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final JsonObject payload = new JsonObject()
```

Comment:
again, you should/could use the factory methods of `CredentialsObject`, I guess.

Reply:
There a couple of more places in this tests that this could be improved